src/Form/Form.js

```diff
@@ -444,13 +417,6 @@ class Form {
             return this
         }
 
-        // When new inputs are added after the initial scan, reanalyze the whole form
-        if (this.initialScanComplete) {
-            this.formAnalyzer = new FormAnalyzer(this.form, input, this.matching)
-            this.recategorizeAllInputs()
-            return this
-        }
```

Comment:
This is what was causing the problem. Basically, in certain edge cases the container of the field (what here we call `this.form`) could become the `body`. When that happened, any new field was hitting this conditional thus dropping everything and starting from scratch. We were doing this for every single added field, which in Asana could be hundreds at a time as you scroll through large projects. This wasn't always hitting the failsafes because we kept destroying and re-building the form data.

In our browser the slowdown was limited (still massive), but on Firefox it could hang for several seconds and show the warning.